pkg/provisioner/ironic/ironic.go

```diff
@@ -1336,6 +1336,25 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if p.host.Status.PoweredOn {
```

Comment:
Should this be looking at the `ironicNode.PowerState` value instead of the version of that copied to the host?

Reply:
Done.